### PR TITLE
Refactored to use poetry instead of setup.py

### PR DESCRIPTION
**Pull request by bendhouseart**
- Preserved entrypoint run.py for direct/unpackaged use
- deleted bids.py as PET has been added to source at niworkflows.utils.bids
- Moved functionality from run.py to extract_tacs.py
- Created cli() method in extract_tacs.py to provide entry point for poetry/package installation in lieu of __name__ == "main"
- bumped version to 0.0.4 to account for minor changes

### CODE REVIEW

File: petprep_extract_tacs/utils/merge_tacs.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import glob
 import shutil
 
-from petprep_extract_tacs.bids import collect_participants
+from petprep_extract_tacs.utils.bids import collect_participants
```
Comment (bendhouseart) [inline petprep_extract_tacs/utils/merge_tacs.py:11]: Swap with niworkflows and remove utils/bids.py entirely

**Comment by bendhouseart** [comment]
Update README too! @bendhouseart

File: pyproject.toml [hunk 2]
```diff
@@ -51,8 +51,11 @@ nipreps = "^1.0"
 black = "^24.4.2"
 pytest = "^8.2.2"
 
+[tool.poetry.scripts]
+petprep_extract_tacs = "petprep_extract_tacs.extract_tacs:cli"
+merge_tacs = "petprep_extract_tacs.utils.merge_tacs:main"
+
```
Comment (bendhouseart) [inline pyproject.toml:57]: Added entry points for `petprep_extract_tacs` and `merge_tacs` following install via pip.

File: pyproject.toml [hunk 3]
```diff
@@ -39,7 +39,7 @@ num2words = "0.5.12"
 pybids = ">=0.16.1"
 sqlalchemy = "1.3.24"
 wrapt = "1.15.0"
-niworkflows = "1.7.9"
+niworkflows = "1.10.2"
```
Comment (bendhouseart) [inline pyproject.toml:42]: Updated `niworkflows` to latest to ensure PET data was included when `utils.bids.collect_participants` and `utils.bids.collect_data` are called.

File: petprep_extract_tacs/utils/merge_tacs.py [hunk 4]
```diff
@@ -167,3 +167,7 @@ def merge_tsvs(*args, **kwargs):
         args.bids_dir,
         subjects=args.subjects,
     )
+
+
+if __name__ == "__main__":
+    main()
```
Comment (bendhouseart) [inline petprep_extract_tacs/utils/merge_tacs.py:173]: Added explicit main function.

File: README.md [hunk 5]
```diff
@@ -124,7 +130,7 @@ docker run -a stderr -a stdout --rm \
 -v /Applications/freesurfer/7.4.1/license.txt:/opt/freesurfer/license.txt \
 --platform linux/amd64 \
 petprep_extract_tacs \
---bids_dir /bids_dir --output_dir /output_dir --analysis_level participant --n_procs 4 system_platform=Darwin
+--bids_dir /bids_dir /output_dir participant --n_procs 4 system_platform=Darwin
```
Comment (bendhouseart) [inline README.md:133]: Removed `--bids_dir` and `--output_dir` flags from README description, added in example of usage following pip install.

File: setup.py [hunk 6]
```diff
@@ -1,16 +0,0 @@
...
-    entry_points={
-        "console_scripts": [
-            "petprep_extract_tacs=petprep_extract_tacs.run:main",
-        ],
-    },
-    url="https://github.com/mnoergaard/petprep_extract_tacs",
-    author="Martin Norgaard",
-    author_email="martin.noergaard@di.ku.dk",
-    description="PETPrep Extraction of Time Activity Curves",
-)
```
Comment (bendhouseart) [inline setup.py:16]: Removed in favor of using `poetry build` instead, this stuff all makes sense to someone, but poetry "just works".

File: version [hunk 7]
```diff
@@ -1 +0,0 @@
-0.0.3
```
Comment (bendhouseart) [inline version:1]: Moved to pyproject.toml.

**Comment by bendhouseart** [comment]
Lastly, had to be more explicit and pass the `args` namespace object to the `init_<wf>` and other such objects following refactor of `petprep_extract_tacs` and move from `args` being global to a local var.

File: petprep_extract_tacs/utils/merge_tacs.py [hunk 8]
```diff
@@ -8,7 +8,7 @@
 import glob
 import shutil
 
-from petprep_extract_tacs.bids import collect_participants
+from niworkflows.utils.bids import collect_participants, collect_data
```
Comment (mnoergaard) [inline petprep_extract_tacs/utils/merge_tacs.py:11]: great

File: setup.py (hunk 6 above)
Comment (mnoergaard) [reply setup.py:16]: Let's do it your way :-)

**Review by mnoergaard — CHANGES_REQUESTED** [review]
Overall, LGTM.

File: README.md (hunk 5 above)
Comment (bendhouseart) [reply README.md]: Weird, must have missed or merged in from main or something silly. Fixed.

**Comment by bendhouseart** [comment]
Corrections applied, thanks for reviewing.

File: README.md [hunk 9]
```diff
@@ -34,10 +34,16 @@ pip install petprep-extract-tacs
 
 ## Quickstart
 
-To get started, you'll need to have your data organized according to the BIDS standard. Once that's in place, you can run the app like this:
+After installation, you'll need to have your data organized according to the BIDS standard. Once that's in place, you can run the app like this:
 
 ```bash
-python3 run.py --bids_dir /path/to/your/bids/dataset --output_dir /path/to/output/dir --n_procs 4 --wm
+petprep_extract_tacs /path/to/your/bids/dataset /path/to/output/dir --n_procs 4 --wm
```
Comment (bendhouseart) [inline README.md]:
```suggestion
petprep_extract_tacs /path/to/your/bids/dataset /path/to/output/dir participant --n_procs 4 --wm
```

Comment (bendhouseart) [reply README.md]: Weird, this looks fixed to me on commit 5fe7baed720291430, but appears s unchanged here on this review page.

**Pull request merged**